### DISCUSSION TITLE
Is there a SVG group equivalent?

### DISCUSSION THREAD

**Discussion by flosse** [emilk/egui#1310] [Q&A]
In SVG I'd place graphical object like this:

```svg
<g transform="translate(50,20)">
  <rect width="50" height="25" rx="5" />
  <line x1="5" y1="8" x2="45" y2="8" />
</g>
```
How can I paint the same with egui?

I started with this but I'm missing the `translate` for the `group`

```rust
use crate::egui::{self, color::Rgba, Stroke};

pub fn test_ui(ui: &mut egui::Ui) -> egui::Response {
    let desired_size = ui.spacing().interact_size.y * egui::vec2(50.0, 25.0);
    let (rect, response) = ui.allocate_exact_size(desired_size, egui::Sense::click());
    if ui.is_rect_visible(rect) {
        ui.group(|ui| {
            ui.painter().rect(
                rect,
                5.0,
                Rgba::WHITE,
                Stroke::new(0.5, Rgba::from_gray(0.1)),
            );
            ui.painter().line_segment(
                [(5.0, 8.0).into(), (45.0, 8.0).into()],
                Stroke::new(1.0, Rgba::from_rgb(255.0, 0.0, 0.0)),
            );
        });
    }
    response
}
```

**Comment by xudesheng** [comment]
I have implemented one but I don't have time to submit it yet. Likely I will submit an example next week or two.
the approach I'm using is to simplify the SVG into `line` and `cubic Bezier curve` only before rendering it. (this is also why I submitted the [PR](https://github.com/emilk/egui/pull/1178).

**Comment by emilk** [comment]
If you want to place an `Ui` you can use https://docs.rs/egui/latest/egui/struct.Ui.html#method.allocate_ui_at_rect or `ui.child_ui` or just `Ui::new`
